docs/src/extend/custom-parsers.md

```diff
@@ -8,63 +8,54 @@ eleventyNavigation:
 
 ---
 
-If you want to use your own parser and provide additional capabilities for your rules, you can specify your own custom parser. If a `parseForESLint` method is exposed on the parser, this method will be used to parse the code. Otherwise, the `parse` method will be used. Both methods should take in the source code as the first argument, and an optional configuration object as the second argument (provided as `parserOptions` in a config file). The `parse` method should simply return the AST. The `parseForESLint` method should return an object that contains the required property `ast` and optional properties `services`, `scopeManager`, and `visitorKeys`.
+ESLint custom parsers let you extend ESLint to support linting new non-standard JavaScript language features or custom syntax in your code. A parser is responsible for taking your code and transforming it into an abstract syntax tree (AST) that ESLint can then analyze and lint.
 
-* `ast` should contain the AST.
-* `services` can contain any parser-dependent services (such as type checkers for nodes). The value of the `services` property is available to rules as `context.parserServices`. Default is an empty object.
-* `scopeManager` can be a [ScopeManager](./scope-manager-interface) object. Custom parsers can use customized scope analysis for experimental/enhancement syntaxes. Default is the `ScopeManager` object which is created by [eslint-scope](https://github.com/eslint/eslint-scope).
-    * Support for `scopeManager` was added in ESLint v4.14.0. ESLint versions which support `scopeManager` will provide an `eslintScopeManager: true` property in `parserOptions`, which can be used for feature detection.
-* `visitorKeys` can be an object to customize AST traversal. The keys of the object are the type of AST nodes. Each value is an array of the property names which should be traversed. Default is [KEYS of `eslint-visitor-keys`](https://github.com/eslint/eslint-visitor-keys#evkkeys).
-    * Support for `visitorKeys` was added in ESLint v4.14.0. ESLint versions which support `visitorKeys` will provide an `eslintVisitorKeys: true` property in `parserOptions`, which can be used for feature detection.
+## Creating a Custom Parser
 
-You can find an ESLint parser project [here](https://github.com/typescript-eslint/typescript-eslint).
+If a `parseForESLint` method is exposed on the parser, this method will be used to parse the code. Otherwise, the `parse` method will be used. Both methods should take in the source code as the first argument, and an optional configuration object as the second argument (provided as `parserOptions` in a config file).
 
-```json
-{
-    "parser": "./path/to/awesome-custom-parser.js"
-}
+```javascript
+// TODO: have a simple example here w parse and parserOptions
```

Comment:
to reviewers, would you mind providing some high level direction on what should be included here? perhaps a simple example with a bit of description?

Reply:
I think that the other [example](https://github.com/eslint/eslint/blob/e6f47cdb2baa5c5153abf16ead00379583003492/docs/src/extend/custom-parsers.md#example) on that page is already quite clear. But you would like to add a new example with `parse` rather than `parseForESLint` if I understand correctly?

Reply:
yes, that's correct

Reply:
You could show just a very simple parser, like one that calls espree and prints a message to the console. This one for example will print the time required to parse each file:

```js
const espree = require("espree");

exports.parse = function (code, options) {
    const label = `Parsing file "${options.filePath}"`;
    console.time(label);
    const ast = espree.parse(code, options);
    console.timeEnd(label);
    return ast; // Only the AST is returned.
};
```

As I understand it, the only difference is that `parse` only returns the AST, whereas `parseForESLint` also returns additional values that let you customize the behavior of the parser even more.

Reply:
> As I understand it, the only difference is that parse only returns the AST, whereas parseForESLint also returns additional values that let you customize the behavior of the parser even more.

That's correct. The usage is exactly the same. I think @fasttime's example is solid -- anything where `parse` is calling another parser and somehow manipulating the AST result before returning it would also be appropriate.

Reply:
thank you for the feedback! also TIL about `console.time()` 💡